packages/node-experimental/src/opentelemetry/spanExporter.ts

```diff
@@ -0,0 +1,301 @@
+import { logger } from '@sentry/utils';
+
+import { OTEL_ATTR_OP, OTEL_ATTR_ORIGIN, OTEL_ATTR_PARENT_SAMPLED, OTEL_ATTR_SOURCE } from '../constants';
+import { getCurrentHub } from '../sdk/hub';
+import { NodeExperimentalScope } from '../sdk/scope';
+import type { NodeExperimentalTransaction } from '../sdk/transaction';
+import { startTransaction } from '../sdk/transaction';
+import type { OtelSpan } from '../types';
+import { convertOtelTimeToSeconds } from '../utils/convertOtelTimeToSeconds';
+import { getRequestSpanData } from '../utils/getRequestSpanData';
+import type { OtelSpanNode } from '../utils/groupOtelSpansWithParents';
+import { groupOtelSpansWithParents } from '../utils/groupOtelSpansWithParents';
+import { getOtelSpanHub, getOtelSpanMetadata, getOtelSpanScope } from './spanData';
+
+/**
+ * A Sentry-specific exporter that converts OpenTelemetry Spans to Sentry Spans & Transactions.
+ */
+export class SentrySpanExporter implements SpanExporter {
+  private _finishedSpans: OtelSpan[];
+  private _stopped: boolean;
+
+  public constructor() {
+    this._stopped = false;
+    this._finishedSpans = [];
+  }
+
+  /** @inheritDoc */
+  public export(spans: OtelSpan[], resultCallback: (result: ExportResult) => void): void {
+    if (this._stopped) {
+      return resultCallback({
+        code: ExportResultCode.FAILED,
+        error: new Error('Exporter has been stopped'),
+      });
+    }
+
+    const openSpanCount = this._finishedSpans.length;
+    const newSpanCount = spans.length;
+
+    this._finishedSpans.push(...spans);
+
+    const remainingSpans = maybeSend(this._finishedSpans);
+
+    const remainingOpenSpanCount = remainingSpans.length;
+    const sentSpanCount = openSpanCount + newSpanCount - remainingOpenSpanCount;
+
+    __DEBUG_BUILD__ &&
+      logger.log(`SpanExporter exported ${sentSpanCount} spans, ${remainingOpenSpanCount} unsent spans remaining`);
+
+    this._finishedSpans = remainingSpans.filter(span => {
+      const shouldDrop = shouldCleanupSpan(span, 5 * 60);
+      __DEBUG_BUILD__ &&
+        shouldDrop &&
+        logger.log(
+          `SpanExporter dropping span ${span.name} (${
+            span.spanContext().spanId
+          }) because it is pending for more than 5 minutes.`,
+        );
+      return !shouldDrop;
+    });
+
+    setTimeout(() => resultCallback({ code: ExportResultCode.SUCCESS }), 0);
+  }
+
+  /** @inheritDoc */
+  public shutdown(): Promise<void> {
+    this._stopped = true;
+    this._finishedSpans = [];
+    return this.forceFlush();
+  }
+
+  /** @inheritDoc */
+  public async forceFlush(): Promise<void> {
+    await flush();
+  }
+}
+
+/** Send the given spans, but only if they are part of a finished transaction. Returns the unsent spans. */
+function maybeSend(spans: OtelSpan[]): OtelSpan[] {
+  const grouped = groupOtelSpansWithParents(spans);
+  const remaining = new Set(grouped);
+
+  const rootNodes = grouped.filter(node => node.span && !node.parentNode);
```

Comment:
```suggestion
  const rootNodes = grouped.filter((node): node is OtelSpan => node.span && !node.parentNode);
```

Reply:
ah, good point, nice! We can do this, but slightly different: 

```ts
const rootNodes = grouped.filter(
    (node): node is OtelSpanNode & { span: OtelSpan } => !!node.span && !node.parentNode,
  );
```